include/boost/geometry/iterators/closing_iterator.hpp

```diff
@@ -38,10 +38,24 @@ struct closing_iterator
     <
         closing_iterator<Range>,
         typename boost::range_value<Range>::type const,
-        boost::random_access_traversal_tag
+        boost::random_access_traversal_tag,
+        typename boost::range_reference<Range>::type const,
```

Comment:
I just carried forward the same style used with boost::range_value above it on line 40, but thinking about it there could be a difference if the range is using a proxy type instead of a regular reference (which I don't think would be an issue with range's value type?). I'll fix it up.

Reply:
I think for value type it should be ok but we'll see. If anything goes wrong the tests on various platforms should detect that.